### DISCUSSION TITLE
HM-ES-TX-WM als Wassersensor nutzbar machen in HA?

### DISCUSSION THREAD

**Discussion by jens-maus** [SukramJ/aiohomematic#873] [Q&A] [answered]
Hallo Zusammen,

seit vielen Jahren verwende ich `HM-ES-TX-WM` Geräte zum Strom- und Gasverbrauch zu messen und hab die auch bereits in HA entsprechend in die Energiemessung integrieren können. Zusätzlich nutze ich auch noch einen dritten `HM-ES-TX-WM` den ich jedoch entsprechend an meinen Wasserzähler angeschlossen habe und in der CCU/RaspberryMatic funktioniert der auch tadellos und erlaubt mir via des "Gas" Datenpunktes den Wasserverbrauch auch in m3 anzuzeigen und auch auswertbar vorzuhalten.

Nun wollte ich diesen `HM-ES-TX-WM` Wassersensor auch im Energiepanel von HomeAssistant integrieren, kann aber in dem Auswahlfenster für den "Wasserverbrauch" die Entität nicht auswählen, vmtl. weil die Datenpunkte eines HM-ES-TX-WM irgendwie intern in der Integration nur für Strom- und Gasmessung angelegt werden und deshalb HA deshalb die Entitäten vorfiltert und quasi keinen Wassersensor dort findet.

In HA sieht dieser wie folgt aus:

<img width="809" alt="Bildschirm­foto 2023-01-20 um 11 27 13" src="https://user-images.githubusercontent.com/5887769/213673470-2215c572-ad56-4915-bbd5-62eb6a882219.png">

Er wird aber wie gesagt in der Auswahl als "Wasserquelle" nicht aufgelistet:

<img width="404" alt="Bildschirm­foto 2023-01-20 um 11 29 17" src="https://user-images.githubusercontent.com/5887769/213673781-b5a1fa68-ad1f-409f-9062-a3bf88fd87fd.png">

Unter Entitäten sind die jedoch zu finden:

<img width="1022" alt="Bildschirm­foto 2023-01-20 um 11 30 47" src="https://user-images.githubusercontent.com/5887769/213674109-7ada92e8-f8f3-4e1f-877c-d78128211959.png">

Gibt es eine Möglichkeit die "Gas Energy Counter" und "Gas Power" Entitäten dieses `HM-ES-TX-WM`-basierten Wassersensors irgendwie zu "markieren" bzw. zu "faken" damit HA mir trotzdem erlaubt diese für die Wasserverbrauchanzeige zu nutzen? Und wenn nicht innerhalb von HA, wäre es dann bitte möglich diese Entitäten in der Integration nicht nur für Gasverbrauch, sondern eben auch für Wasserverbrauch irgendwie zu markieren damit HA dann es zulässt diese damit zu verknüpfen.

**Comment by SukramJ** [comment]
Per [customize](https://www.home-assistant.io/docs/configuration/customizing-devices/#customizing-entities) würde ich mal folgende Werte für die Entität setzten:

```
state_class: total_increasing 
unit_of_measurement: m³ 
device_class: water
```
`total_increasing` wenn der der Wert stetig steigt sonst [HA Doku](https://developers.home-assistant.io/docs/core/entity/sensor#available-state-classes).

Bei den Statistiken kann es dauern bis die neuen Werte ausgegeben werden, also warten.
Eventuell  gibt es unter `Entwicklerwerkzeuge -> Statistik` einen Fehler der korrigiert werden muss.

**Comment by jens-maus** [comment, marked as answer]
Ok, nach weiteren Recherchen und Tests konnte ich das ganze nun via `customize:` eintrag selbst in der `configuration.yaml` von HomeAssistant wie folgt lösen:

```yaml
# homeassistant section
homeassistant:
  customize:
    sensor.hausanschluss_wasser_gas_energy_counter:
      device_class: water
    sensor.hausanschluss_wasser_gas_power:
      device_class: water
```

D.h. ich überschreibe einfach die device class dieser beiden Entitäten und setze sie auf `water`. Damit kann man diese dann auch bei Wasserverbrauch nutzen und sie funktionieren dann auch entsprechend.

Die einzige Schlussfrage wäre noch ob man hier nicht ggf. in der Integration selbst bereits diese Entitäten mehr als einer device class zuordnen kann, dann würde das out-of-the-box funktionieren?

**Reply by SukramJ** [reply]
> Die einzige Schlussfrage wäre noch ob man hier nicht ggf. in der Integration selbst bereits diese Entitäten mehr als einer device class zuordnen paar machen kann?

Nein. Eine Entität hat immer eine `device_class` mit der passenden `unit_of_measurement` und ggf. eine `state_class`

**Reply by jens-maus** [reply]
Schade. Vielleicht wäre das ja eine Sache für ein mögliches Enhancement Request für HA. An diesem Beispiel sieht man ja das anscheinend eine Liste von device classes nicht ganz unsinnig sein kann, weil ein Gerät (muss ja nicht Homematic sein) ggf. zwei Nutzungsszenarien haben kann und damit zwei oder mehreren device classes zugesprochen werden kann. Da sollte die Integration IMHO in der Lage sein diese beiden Nutzungsszenarien eben so in der Angabe der `device_class` auch umzusetzen. Dann könnte der Nutzer das Gerät bzw. die Entität entsprechend auswählen.

Oder wäre es alternativ möglich in der Integration einfach für den `HM-ES-TX-WM` zwei weitere Entitäten hinzuzufügen die quasi Kopien der `gas_energy_counter` und `gas_power` entitäten sind, jedoch eben `water_XXXX`  heissen und der `water` device class zugeordnet sind und einfach 1:1 die gleichen werte dann mitbringen/aktualisieren wie die `gas_energy_counter`, etc.? Das Gerät hat ja ohnehin auch parallel Strom-Datenpunkte/Entitäten auch wenn diese nicht genutzt werden wenn dieser als Gaszähler eingesetzt wird.

**Reply by SukramJ** [reply]
Mit deinem Vorschlag für die Integration kannst Du mich nicht begeistern, denn für genau solche Fälle gibt es in HA die Möglichkeit ein Customizing durchzuführen. 

Wenn ich jedoch in einem Gerät einen Datenpunkt (MASTER /VALUES) habe, der die entsprechende Konfiguration enthält, kann ich in der Integration darauf reagieren. Das mache ich u.a. für den CHANNEL_OPERATION_MODE um das Verhalten von Eingängen (Taster, Schalter, Sensor) entsprechend auf Entitäten abbilden zu können.

**Comment by RacingHell** [comment]
Wäre das Vorgehen in diesem speziellen Fall denn nicht an geeigneter Stelle einen “googlebaren” Wiki-Eintrag wert? Ich hatte die selbe Problemstellung wie jens-maus und für mich gesprochen als bekennender Laie, der sich durch die Materie durchboxt, echte Mühe die Lösung irgendwo in den Weiten des www’s zu recherchieren.

LG

**Reply by SukramJ** [reply]
Das hört sich gut an. Im [Wiki](https://github.com/danielperna84/hahomematic/wiki) kannst du diese Informationen selber ablegen.
